docs/tutorials/gluon/info_gan.md

```diff
@@ -0,0 +1,438 @@
+if os.path.isfile("infogan_d_latest.params") and os.path.isfile("infogan_g_latest.params"):
+    discriminator.load_parameters('infogan_d_latest.params', ctx=ctx, allow_missing=True, ignore_extra=True)
+    generator.load_parameters('infogan_g_latest.params', ctx=ctx, allow_missing=True, ignore_extra=True)
+```
+
+The latent code `c` is part of the Generator input and it contains mutliple variables (continuous, categorical) that can represent different distributions. In order to make sure that the Generator uses the latent code, mutual information is introduced into the GAN loss term. Mutual information measures how much X is known given Y or vice versa. It is defined as:
+
+![gif](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/entropy.gif) 
+
+The InfoGAN loss is:
+
+![gif](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/loss.gif)
+
+where `V(D,G)` is the GAN loss and the mutual information `I(c, G(z, c))` goes in as regularization. The goal is to reach high mutual information, in order to learn meaningful codes `c` for the data. 
+
+
+Define the loss functions. `SoftmaxCrossEntropyLoss` for the categorical code `c`,  `L2Loss` for the continious code `c` and `SigmoidBinaryCrossEntropyLoss` for the normal GAN loss.
+
+
+```python
+loss1 = gluon.loss.SigmoidBinaryCrossEntropyLoss()
+loss2 = gluon.loss.L2Loss()
+loss3 = gluon.loss.SoftmaxCrossEntropyLoss()
+```
+
+This function samples `c`, `z`, and concatenates them to create the generator input.
+
+
+```python
+def create_generator_input():
+    
+    #create random noise
+    z      = nd.random_normal(0, 1, shape=(batch_size, z_dim), ctx=ctx)
+    label  = nd.array(np.random.randint(n_categories, size=batch_size)).as_in_context(ctx)
+    c1     = nd.one_hot(label, depth=n_categories).as_in_context(ctx)
+    c2     = nd.random.uniform(-1, 1, shape=(batch_size, n_continuous)).as_in_context(ctx)
+
+    # concatenate random noise with c which will be the input of the generator
+    return nd.concat(z, c1, c2, dim=1)
+```
+
+Define the training loop. 
+1. The discriminator receives `real_data` and `loss1` measures how many real images have been identified as real
+2. The discriminator receives `fake_image` from the Generator and `loss1` measures how many fake images have been identified as fake
+3. Update Discriminator. Currently, it is updated every second iteration in order to avoid that the Discriminator becomes too strong. You may want to change that.
+4. The updated discriminator receives `fake_image` and `loss1` measures how many fake images have been been identified as real, `loss2` measures the difference between the sampled continuous latent code `c` and the output of the Q model and `loss3` measures the difference between the sampled categorical latent code `c` and the output of the Q model.
+4. Update Generator and Q
+
+
+```python
+with SummaryWriter(logdir='./logs/') as sw:
+    
+    epochs = 1
+    i = 0
```

Comment:
better name for i

also, you don't have to i+=1 and then later check % 2, you can i = (i + 1) % 2 because that's all you care about. it's a nit, but cleaner as i doesn't increase without bound unnecessarily.